astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
     When subclassing, be sure to list the superclasses in the correct order
```

Comment:
It is probably worth checking what makes it required to have a certain order. Often, one can write things such that order does not matter (but certainly not always).